geodiff/src/sqlitedriver.cpp

```diff
@@ -0,0 +1,576 @@
+/*
+ GEODIFF - MIT License
+ Copyright (C) 2020 Martin Dobias
+*/
+
+#include "sqlitedriver.h"
+
+#include "geodiffchangeset.h"
+
+#include <memory.h>
+
+void SqliteDriver::open( const std::map<std::string, std::string> &conn )
+{
+  auto connBaseIt = conn.find( "base" );
+  if ( connBaseIt == conn.end() )
+    throw GeoDiffException( "Missing 'base' file" );
+
+  auto connModifiedIt = conn.find( "modified" );
+  mHasModified = connModifiedIt != conn.end();
+
+  std::string base = connBaseIt->second;
+
+  db = std::make_shared<Sqlite3Db>();
+  if ( mHasModified )
+  {
+    db->open( connModifiedIt->second );
+
+    Buffer sqlBuf;
+    sqlBuf.printf( "ATTACH '%s' AS aux", base.c_str() );
+    db->exec( sqlBuf );
+  }
+  else
+  {
+    db->open( base );
+  }
+
+  // GeoPackage triggers require few functions like ST_IsEmpty() to be registered
+  // in order to be able to apply changesets
+  if ( isGeoPackage( db ) )
+  {
+    bool success = register_gpkg_extensions( db );
+    if ( !success )
+    {
+      throw GeoDiffException( "Unable to enable sqlite3/gpkg extensions" );
+    }
+  }
+
+}
+
+void SqliteDriver::listTables( std::vector<std::string> &tableNames, bool useModified )
```

Comment:
would it be more convinient to return std::vector of tablenames instead of using it in the argument?